tests/acceptance/helpers/sharingHelper.js

```diff
@@ -111,28 +105,55 @@ module.exports = {
     const apiURL = client.globals.backend_url + '/ocs/v2.php/apps/files_sharing/api/v1/shares?format=json'
     let lastShareToken
+    let lastShare
     await fetch(apiURL, { method: 'GET', headers: headers })
       .then(res => res.json())
       .then(function (sharesResult) {
-        if (sharesResult.ocs.meta.statuscode === 200) {
-          const shares = sharesResult.ocs.data
-          let lastFoundShareId = 0
-          for (var shareI = 0; shareI < shares.length; shareI++) {
-            const share = shares[shareI]
-            if (share.share_type === self.SHARE_TYPES.public_link && share.id >= lastFoundShareId) {
-              lastFoundShareId = share.id
-              lastShareToken = share.token
-            }
+        httpHelper.checkOCSStatus(sharesResult, 'Could not get shares. Message: ' + sharesResult.ocs.meta.message)
+        const shares = sharesResult.ocs.data
+        let lastFoundShareId = 0
+        for (const share of shares) {
+          if (share.share_type === self.SHARE_TYPES.public_link && share.id >= lastFoundShareId) {
+            lastFoundShareId = share.id
+            lastShareToken = share.token
+            lastShare = share
           }
-          if (lastShareToken === null) {
-            throw Error('Could not find public shares. All shares: ' + JSON.stringify(shares, null, 2))
+        }
+        if (lastShareToken === null) {
+          throw Error('Could not find public shares. All shares: ' + JSON.stringify(shares, null, 2))
+        }
+      })
+    return lastShare
+  },
+  deleteLastPublicLinkShare: async function (sharer, shareId) {
+    const headers = httpHelper.createAuthHeader(sharer)
+    const apiURL = client.globals.backend_url + '/ocs/v2.php/apps/files_sharing/api/v1/shares/' + shareId + '?format=json'
+    await fetch(apiURL, { method: 'DELETE', headers: headers })
+      .then(res => res.json())
+      .then(function (response) {
+        httpHelper.checkOCSStatus(response, "Couldn't delete share. Message: " + response.ocs.meta.message)
+      })
+  },
+  /**
+   * function that returns array of public link shares created on the name of a user
+   * @param sharer user whose all public links are to be fetched
+   * @returns array of public link shares of that sharer
+   */
+  getAllPublicLinkShares: async function (sharer) {
+    const headers = httpHelper.createAuthHeader(sharer)
+    const data = []
+    const apiURL = client.globals.backend_url + '/ocs/v2.php/apps/files_sharing/api/v1/shares?&format=json'
+    await fetch(apiURL, { method: 'GET', headers: headers })
+      .then(res => res.json())
+      .then(res => {
+        httpHelper.checkOCSStatus(res, 'Could not get shares. Message: ' + res.ocs.meta.message)
+        for (const share of res.ocs.data) {
+          if (share.share_type === 3 && share.uid_owner === sharer) {
+            data.push(share)
           }
-        } else {
-          throw Error('Could not get shares. Message: ' + sharesResult.ocs.meta.message)
```

Comment:
Why was this deleted?

Reply:
It's because error is thrown by checkOCSStatus() function. So, there is no need of checking again.